src/client.ts

```diff
@@ -71,28 +35,30 @@ export interface GenerationOptions {
 }
 
 export class Session {
-  constructor(private connectionString: string) {
-    const info = mkUrl("run", connectionString);
-    if (!(info.url.startsWith("http://") || info.url.startsWith("https://")))
-      throw new Error("Invalid URL: " + connectionString);
-    if (Object.keys(info.headers).length == 0)
-      throw new Error("No key in connection string");
-  }
+  private oai_client: OpenAI;
+  private model: string;
 
-  resolvePath(url: string) {
-    return mkUrl(url, this.connectionString);
+  constructor(baseUri: string, targetModel: string) {
+    this.oai_client = new OpenAI({ baseURL: baseUri, apiKey: "" });
+    this.model = targetModel;
   }
 
   async request(options: RequestOptions) {
-    const info = this.resolvePath(options.url);
-    return await postAndRead({
-      ...options,
-      url: info.url,
-      headers: {
-        ...info.headers,
-        ...(options.headers ?? {}),
-      },
-    });
+    console.log("Options.data:", options.data);
+    console.log("Messages:", options.data?.messages);
+    console.log("guided_grammar:", options.data?.grammar);
+    const response = await this.oai_client.chat.completions.create({
+      model: this.model,
+      messages: options.data?.messages ?? [],
+    }, /*{
+      extra_body: {
```

Comment:
This is what the Python code uses. From the name, I guess it's appending to the JSON in the request body